server/game/dudecard.js

```diff
@@ -234,7 +234,9 @@ class DudeCard extends DrawCard {
         this.action({
             title: 'Trade',
             abilitySourceType: 'game',
-            condition: () => this.game.currentPhase === 'high noon' && this.hasAttachmentForTrading(),
+            condition: () => this.game.currentPhase === 'high noon' && 
+                this.isInControlledLocation() &&
```

Comment:
I'd suggest not putting mechanic fixes in the same PR where we add cards.

Reply:
I basically agree with you but at this stage it just slows me down. Many times, I am working on a card and during testing find some small thing so just fix it there. To create new branch, separate the change, then new PR is in my opinion at this time unnecessary. What I try to do is describe the additional change in the commit message.
Once the code will be more stable, then I definitely want to make it separate.